### PR TITLE
Feature 479075: Search (title) shared types and schema

### PR DESCRIPTION
**Pull request by mokhld**
## Story Link
https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/479075

## Description
This PR introduces the shared types and schemas needed for the search addition for `GET /forms` in forms-manager.

### CODE REVIEW

File: model/src/common/search/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Joi from 'joi'
+
+import { type SearchOptions } from '~/src/common/search/types.js'
+
+/**
+ * Field definitions for search options.
+ */
+export const searchOptionFields = {
+  title: Joi.string().trim().min(1).optional()
```
Comment (colinrotherham) [inline model/src/common/search/index.ts:9]: I like seeing `.trim()`, reminds me we should do it in a lot more places

File: model/src/common/search/index.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+/**
+ * Joi schema for `SearchOptions` interface
+ * @see {@link SearchOptions}
```
Comment (colinrotherham) [inline model/src/common/search/index.ts:14]: Can stick the link in the description if you like?

It lets you do Markdown style link text `{@link SearchOptions | Link text here}` too

```suggestion
 * Joi schema for {@link SearchOptions} interface
```

File: model/src/common/search/index.ts (hunk 1 above)
Comment (colinrotherham) [inline model/src/common/search/index.ts:9]: Can I double check what "optional" means in a GET sense?

Think we also need `.allow('')` for when `?title=` is an empty string

**Review by colinrotherham — APPROVED** [review]

File: model/src/common/search/index.ts (hunk 2 above)
Comment (mokhld) [reply model/src/common/search/index.ts:14]: That's more concise for sure.

File: model/src/common/search/index.ts (hunk 1 above)
Comment (mokhld) [reply model/src/common/search/index.ts:9]: That's a good spot! Joi wouldn't let an empty string through- thought it would as it's optional

Comment (mokhld) [reply model/src/common/search/index.ts:9]: I think we can just remove `min(1)`

Comment (mokhld) [reply model/src/common/search/index.ts:9]: Nope I was wrong 😅

**Pull request merged**